judge/views/submission.py

```diff
@@ -330,16 +336,20 @@ def get_context_data(self, **kwargs):
 
 
 class AllUserSubmissions(ConditionalUserTabMixin, UserMixin, SubmissionsListBase):
+    @cached_property
+    def is_own(self):
```

Comment:
Since this code is duplicated in `UserProblemSubmissions`, maybe we should bring this into `ConditionalUserTabMixin` (there is another instance of `self.request.user.is_authenticated and self.request.profile == self.profile` in that class).